README.md
# eslint-config-liting

an eslint preset of liting-yes

# support

- JavaScript
- TypeScript
- Vue.js

# Precaution

- By default, vue3 is used. If you are using vue2, you need to override the default rules
  ```json
  {
    "extends": [
      "liting",
      "eslint-config-liting/src/vue2"
    ],
  }
  ```


CHANGELOG.md
# Changelog

## [0.3.1](https://github.com/dev-standard/eslint-config-liting/compare/0.3.0...0.3.1) (2022-12-17)


### Features

* **test:** init ([65f6cd6](https://github.com/dev-standard/eslint-config-liting/commit/65f6cd6c75afc2c66ed0f2462e321cd6935ebddc))

## [0.3.0](https://github.com/dev-standard/eslint-config-liting/compare/0.2.1...0.3.0) (2022-12-16)


### Features

* support ts ([b1319e0](https://github.com/dev-standard/eslint-config-liting/commit/b1319e0598f4be358e676eee916c608d97a203b7))

## [0.2.1](https://github.com/dev-standard/eslint-config-liting/compare/0.2.0...0.2.1) (2022-12-13)

## [0.2.0](https://github.com/dev-standard/eslint-config-liting/compare/0.1.0...0.2.0) (2022-12-13)


### Features

* support vue rules ([0dd2e58](https://github.com/dev-standard/eslint-config-liting/commit/0dd2e58769f7c64f6075507dcd8a99e89d72960f))

## 0.1.0 (2022-12-07)

__tests__/test.js
/*
 * @Date: 2022-12-17 14:33:25
 * @Author: liting luz.liting@gmail.com
 * @LastEditors: liting luz.liting@gmail.com
 * @LastEditTime: 2022-12-17 14:35:49
 * @FilePath: /eslint-config-liting/__tests__/test.js
 */
const testJs = 'javascript'


__tests__/test.ts
/*
 * @Date: 2022-12-17 14:34:33
 * @Author: liting luz.liting@gmail.com
 * @LastEditors: liting luz.liting@gmail.com
 * @LastEditTime: 2022-12-17 14:36:52
 * @FilePath: /eslint-config-liting/__tests__/test.ts
 */

type Test = string
const testTs: Test = 'typescript'


src/vue.js
module.exports = {
  overrides: [
    {
      extends: ['plugin:vue/vue3-recommended'],
      files: ['*.vue'],
      parser: 'vue-eslint-parser',
      parserOptions: {
        parser: {
          js: 'espree',
          ts: '@typescript-eslint/parser',
        },
        sourceType: 'module',
      },
    },
  ],
  rules: {
    indent: 'off',
    // base
    'vue/comment-directive': [
      'error',
      {
        reportUnusedDisableDirectives: true,
      },
    ],
    'vue/jsx-uses-vars': 'error',

    // essential
    'vue/multi-word-component-names': 'error',
    'vue/no-arrow-functions-in-watch': 'error',
    'vue/no-async-in-computed-properties': 'error',
    'vue/no-child-content': 'error',
    'vue/no-computed-properties-in-data': 'error',
    'vue/no-dupe-keys': 'error',
    'vue/no-dupe-v-else-if': 'error',
    'vue/no-duplicate-attributes': [
      'error',
      {
        allowCoexistClass: true,
        allowCoexistStyle: true,
      },
    ],
    'vue/no-export-in-script-setup': 'error',
    'vue/no-mutating-props': 'error',
    'vue/no-parsing-error': [
      'error',
      {
        'abrupt-closing-of-empty-comment': true,
        'absence-of-digits-in-numeric-character-reference': true,
        'cdata-in-html-content': true,
        'character-reference-outside-unicode-range': true,
        'control-character-in-input-stream': true,
        'control-character-reference': true,
        'duplicate-attribute': true,
        'end-tag-with-attributes': true,
        'end-tag-with-trailing-solidus': true,
        'eof-before-tag-name': true,
        'eof-in-cdata': true,
        'eof-in-comment': true,
        'eof-in-tag': true,
        'incorrectly-closed-comment': true,
        'incorrectly-opened-comment': true,
        'invalid-first-character-of-tag-name': true,
        'missing-attribute-value': true,
        'missing-end-tag-name': true,
        'missing-semicolon-after-character-reference': true,
        'missing-whitespace-between-attributes': true,
        'nested-comment': true,
        'non-void-html-element-start-tag-with-trailing-solidus': false,
        'noncharacter-character-reference': true,
        'noncharacter-in-input-stream': true,
        'null-character-reference': true,
        'surrogate-character-reference': true,
        'surrogate-in-input-stream': true,
        'unexpected-character-in-attribute-name': true,
        'unexpected-character-in-unquoted-attribute-value': true,
        'unexpected-equals-sign-before-attribute-name': true,
        'unexpected-null-character': true,
        'unexpected-question-mark-instead-of-tag-name': true,
        'unexpected-solidus-in-tag': true,
        'unknown-named-character-reference': true,
        'x-invalid-end-tag': true,
        'x-invalid-namespace': true,
      },
    ],
    'vue/no-ref-as-operand': 'error',
    'vue/no-reserved-component-names': [
      'error',
      {
        disallowVue3BuiltInComponents: false,
        disallowVueBuiltInComponents: false,
      },
    ],
    'vue/no-reserved-keys': 'error',
    'vue/no-reserved-props': [
      'error',
      {
        vueVersion: 3,
      },
    ],
    'vue/no-setup-props-destructure': 'error',
    'vue/no-shared-component-data': 'error',
    'vue/no-side-effects-in-computed-properties': 'error',
    'vue/no-template-key': 'error',
    'vue/no-textarea-mustache': 'error',
    'vue/no-unused-components': [
      'error',
      {
        ignoreWhenBindingPresent: true,
      },
    ],
    'vue/no-unused-vars': 'error',
    'vue/no-use-computed-property-like-method': 'error',
    'vue/no-use-v-if-with-v-for': [
      'error',
      {
        allowUsingIterationVar: false,
      },
    ],
    'vue/no-useless-template-attributes': 'error',
    'vue/no-v-text-v-html-on-component': 'error',
    'vue/require-component-is': 'error',
    'vue/require-prop-type-constructor': 'error',
    'vue/require-render-return': 'error',
    'vue/require-v-for-key': 'error',
    'vue/require-valid-default-prop': 'error',
    'vue/return-in-computed-property': [
      'error',
      {
        treatUndefinedAsUnspecified: true,
      },
    ],
    'vue/return-in-emits-validator': 'error',
    'vue/use-v-on-exact': 'error',
    'vue/valid-attribute-name': 'error',
    'vue/valid-define-emits': 'error',
    'vue/valid-define-props': 'error',
    'vue/valid-next-tick': 'error',
    'vue/valid-template-root': 'error',
    'vue/valid-v-bind': 'error',
    'vue/valid-v-cloak': 'error',
    'vue/valid-v-else': 'error',
    'vue/valid-v-else-if': 'error',
    'vue/valid-v-for': 'error',
    'vue/valid-v-html': 'error',
    'vue/valid-v-if': 'error',
    'vue/valid-v-model': 'error',
    'vue/valid-v-on': 'error',
    'vue/valid-v-once': 'error',
    'vue/valid-v-pre': 'error',
    'vue/valid-v-show': 'error',
    'vue/valid-v-slot': [
      'error',
      {
        allowModifiers: false,
      },
    ],
    'vue/valid-v-text': 'error',

    // essential for Vue.js 3.x
    // eslint-disable-next-line
    'vue/no-deprecated-data-object-declaration': 'error',
    'vue/no-deprecated-destroyed-lifecycle': 'error',
    'vue/no-deprecated-dollar-listeners-api': 'error',
    'vue/no-deprecated-dollar-scopedslots-api': 'error',
    'vue/no-deprecated-events-api': 'error',
    'vue/no-deprecated-filter': 'error',
    'vue/no-deprecated-functional-template': 'error',
    'vue/no-deprecated-html-element-is': 'error',
    'vue/no-deprecated-inline-template': 'error',
    'vue/no-deprecated-props-default-this': 'error',
    'vue/no-deprecated-router-link-tag-prop': [
      'error',
      {
        components: ['RouterLink'],
      },
    ],
    'vue/no-deprecated-scope-attribute': 'error',
    'vue/no-deprecated-slot-attribute': 'error',
    'vue/no-deprecated-slot-scope-attribute': 'error',
    'vue/no-deprecated-v-bind-sync': 'error',
    'vue/no-deprecated-v-is': 'error',
    'vue/no-deprecated-v-on-native-modifier': 'error',
    'vue/no-deprecated-v-on-number-modifiers': 'error',
    'vue/no-deprecated-vue-config-keycodes': 'error',
    'vue/no-expose-after-await': 'error',
    'vue/no-lifecycle-after-await': 'error',
    'vue/no-v-for-template-key-on-child': 'error',
    'vue/no-watch-after-await': 'error',
    'vue/prefer-import-from-vue': 'error',
    'vue/require-slots-as-functions': 'error',
    'vue/require-toggle-inside-transition': 'error',
    'vue/valid-v-is': 'error',
    'vue/valid-v-memo': 'error',

    // strongly Recommended
    // eslint-disable-next-line
    'vue/attribute-hyphenation': ['error', 'always'],
    'vue/component-definition-name-casing': ['error', 'PascalCase'],
    'vue/first-attribute-linebreak': [
      'error',
      {
        multiline: 'below',
        singleline: 'beside',
      },
    ],
    'vue/html-closing-bracket-newline': [
      'error',
      {
        multiline: 'always',
        singleline: 'never',
      },
    ],
    'vue/html-closing-bracket-spacing': [
      'error',
      {
        endTag: 'never',
        selfClosingTag: 'always',
        startTag: 'never',
      },
    ],
    'vue/html-end-tags': 'error',
    'vue/html-indent': [
      'error',
      // eslint-disable-next-line
      2,
      {
        alignAttributesVertically: true,
        attribute: 1,
        baseIndent: 1,
        closeBracket: 0,
      },
    ],
    'vue/html-quotes': [
      'error',
      'single',
      {
        avoidEscape: true,
      },
    ],
    'vue/html-self-closing': [
      'error',
      {
        html: {
          component: 'always',
          normal: 'always',
          void: 'never',
        },
        math: 'always',
        svg: 'always',
      },
    ],
    'vue/max-attributes-per-line': [
      'error',
      {
        multiline: 1,
        singleline: 3,
      },
    ],
    'vue/multiline-html-element-content-newline': [
      'error',
      {
        allowEmptyLines: false,
        ignoreWhenEmpty: true,
      },
    ],
    'vue/mustache-interpolation-spacing': ['error', 'always'],
    'vue/no-multi-spaces': [
      'error',
      {
        ignoreProperties: false,
      },
    ],
    'vue/no-spaces-around-equal-signs-in-attribute': 'error',
    'vue/no-template-shadow': 'error',
    'vue/one-component-per-file': 'error',
    'vue/prop-name-casing': ['error', 'camelCase'],
    'vue/require-default-prop': 'warn',
    'vue/require-prop-types': 'warn',
    'vue/singleline-html-element-content-newline': [
      'error',
      {
        ignoreWhenEmpty: true,
        ignoreWhenNoAttributes: true,
      },
    ],
    'vue/v-bind-style': ['error', 'shorthand'],
    'vue/v-on-style': ['error', 'shorthand'],
    'vue/v-slot-style': [
      'error',
      {
        atComponent: 'v-slot',
        default: 'shorthand',
        named: 'shorthand',
      },
    ],

    // strongly Recommended for Vue.js 3.x
    // eslint-disable-next-line
    'vue/require-explicit-emits': [
      'error',
      {
        allowProps: false,
      },
    ],
    'vue/v-on-event-hyphenation': [
      'error',
      'always',
      {
        autofix: false,
        ignore: [],
      },
    ],

    // recommended
    // eslint-disable-next-line
    'vue/attributes-order': [
      'error',
      {
        alphabetical: false,
        order: [
          'DEFINITION',
          'LIST_RENDERING',
          'CONDITIONALS',
          'RENDER_MODIFIERS',
          'GLOBAL',
          ['UNIQUE', 'SLOT'],
          'TWO_WAY_BINDING',
          'OTHER_DIRECTIVES',
          'OTHER_ATTR',
          'EVENTS',
          'CONTENT',
        ],
      },
    ],
    'vue/component-tags-order': [
      'error',
      {
        order: [['script', 'template'], 'style'],
      },
    ],
    'vue/no-lone-template': [
      'error',
      {
        ignoreAccessible: true,
      },
    ],
    'vue/no-multiple-slot-args': 'error',
    'vue/no-v-html': 'warn',
    'vue/order-in-components': [
      'error',
      {
        order: [
          'el',
          'name',
          'key',
          'parent',
          'functional',
          ['delimiters', 'comments'],
          ['components', 'directives', 'filters'],
          'extends',
          'mixins',
          ['provide', 'inject'],
          'ROUTER_GUARDS',
          'layout',
          'middleware',
          'validate',
          'scrollToTop',
          'transition',
          'loading',
          'inheritAttrs',
          'model',
          ['props', 'propsData'],
          'emits',
          'setup',
          'asyncData',
          'data',
          'fetch',
          'head',
          'computed',
          'watch',
          'watchQuery',
          'LIFECYCLE_HOOKS',
          'methods',
          ['template', 'render'],
          'renderError',
        ],
      },
    ],
    'vue/this-in-template': 'error',

    // uncategorized
    // eslint-disable-next-line
    'vue/block-lang': [
      'error',
      {
        script: {
          lang: 'ts',
        },
      },
    ],
    'vue/block-tag-newline': [
      'error',
      {
        blocks: {
          'my-block': {
            maxEmptyLines: 0,
            multiline: 'always',
            singleline: 'always',
          },
          script: {
            maxEmptyLines: 0,
            multiline: 'always',
            singleline: 'always',
          },
          template: {
            maxEmptyLines: 0,
            multiline: 'always',
            singleline: 'always',
          },
        },
      },
    ],
    'vue/component-api-style': ['error', ['script-setup', 'composition']],
    'vue/component-name-in-template-casing': [
      'error',
      'PascalCase',
      {
        globals: ['RouterView', 'RouterLink'],
        registeredComponentsOnly: true,
      },
    ],
    'vue/component-options-name-casing': ['error', 'PascalCase'],
    'vue/custom-event-name-casing': ['error', 'kebab-case'],
    'vue/define-emits-declaration': ['error', 'type-based'],
    'vue/define-macros-order': [
      'error',
      {
        order: ['defineProps', 'defineEmits'],
      },
    ],
    'vue/define-props-declaration': ['error', 'type-based'],
    'vue/html-button-has-type': [
      'error',
      {
        button: true,
        reset: true,
        submit: true,
      },
    ],
    'vue/html-comment-content-newline': [
      'error',
      {
        multiline: 'always',
        singleline: 'never',
      },
    ],
    'vue/html-comment-content-spacing': ['error', 'always'],
    'vue/html-comment-indent': ['error', 2],
    'vue/match-component-file-name': [
      'error',
      {
        extensions: ['jsx', 'tsx', 'vue'],
        shouldMatchCase: true,
      },
    ],
    'vue/match-component-import-name': 'error',
    'vue/new-line-between-multi-line-property': [
      'error',
      {
        minLineOfMultilineProperty: 2,
      },
    ],
    'vue/next-tick-style': ['error', 'promise'],
    'vue/no-bare-strings-in-template': [
      'warn',
      {
        allowlist: [
          '(',
          ')',
          ',',
          '.',
          '&',
          '+',
          '-',
          '=',
          '*',
          '/',
          '#',
          '%',
          '!',
          '?',
          ':',
          '[',
          ']',
          '{',
          '}',
          '<',
          '>',
          '\u00b7',
          '\u2022',
          '\u2010',
          '\u2013',
          '\u2014',
          '\u2212',
          '|',
        ],
        attributes: {
          '/.+/': ['title', 'aria-label', 'aria-placeholder', 'aria-roledescription', 'aria-valuetext'],
          img: ['alt'],
          input: ['placeholder'],
        },
        directives: ['v-text'],
      },
    ],
    'vue/no-boolean-default': ['error', 'no-default'],
    'vue/no-duplicate-attr-inheritance': 'error',
    'vue/no-empty-component-block': 'error',
    'vue/no-multiple-objects-in-class': 'error',
    'vue/no-potential-component-option-typo': [
      'error',
      {
        custom: [],
        presets: ['vue'],
        threshold: 1,
      },
    ],
    'vue/no-ref-object-destructure': 'error',
    'vue/no-required-prop-with-default': [
      'error',
      {
        autofix: true,
      },
    ],
    'vue/no-restricted-block': 'off',
    'vue/no-restricted-call-after-await': 'off',
    'vue/no-restricted-class': 'off',
    'vue/no-restricted-component-options': 'off',
    'vue/no-restricted-custom-event': 'off',
    'vue/no-restricted-html-elements': 'off',
    'vue/no-restricted-props': 'off',
    'vue/no-restricted-static-attribute': 'off',
    'vue/no-restricted-v-bind': 'off',
    'vue/no-static-inline-styles': [
      'error',
      {
        allowBinding: false,
      },
    ],
    'vue/no-template-target-blank': [
      'error',
      {
        allowReferrer: false,
        enforceDynamicLinks: 'always',
      },
    ],
    'vue/no-this-in-before-route-enter': 'error',
    'vue/no-undef-components': 'error',
    'vue/no-undef-properties': 'error',
    'vue/no-unsupported-features': [
      'error',
      {
        version: '^3.0.0',
      },
    ],
    'vue/no-unused-properties': [
      'error',
      {
        deepData: false,
        groups: ['props', 'data', 'computed', 'methods'],
        ignorePublicMembers: false,
      },
    ],
    'vue/no-unused-refs': 'error',
    'vue/no-useless-mustaches': [
      'error',
      {
        ignoreIncludesComment: false,
        ignoreStringEscape: false,
      },
    ],
    'vue/no-useless-v-bind': [
      'error',
      {
        ignoreIncludesComment: false,
        ignoreStringEscape: false,
      },
    ],
    'vue/no-v-text': 'error',
    'vue/padding-line-between-blocks': ['error', 'always'],
    'vue/padding-line-between-tags': [
      'error',
      [
        {
          blankLine: 'never',
          next: '*',
          prev: '*',
        },
      ],
    ],
    'vue/prefer-prop-type-boolean-first': 'error',
    'vue/prefer-separate-static-class': 'error',
    'vue/prefer-true-attribute-shorthand': ['error', 'always'],
    'vue/require-direct-export': [
      'error',
      {
        disallowFunctionalComponentFunction: false,
      },
    ],
    'vue/require-emit-validator': 'error',
    'vue/require-name-property': 'error',
    'vue/require-prop-comment': 'error',
    'vue/script-indent': [
      'error',
      2,
      {
        baseIndent: 0,
        switchCase: 0,
      },
    ],
    'vue/sort-keys': [
      'error',
      'asc',
      {
        caseSensitive: true,
        ignoreChildrenOf: ['model'],
        ignoreGrandchildrenOf: ['computed', 'directives', 'inject', 'props', 'watch'],
        minKeys: 2,
        natural: false,
      },
    ],
    'vue/static-class-names-order': 'error',
    'vue/v-for-delimiter-style': ['error', 'of'],
    'vue/v-on-handler-style': [
      'error',
      ['method', 'inline-function'],
      {
        ignoreIncludesComment: false,
      },
    ],

    // extension Rules
    // eslint-disable-next-line
    'vue/array-bracket-newline': [
      'error',
      {
        minItems: 2,
        multiline: true,
      },
    ],
    'vue/array-bracket-spacing': [
      'error',
      'always',
      {
        arraysInArrays: true,
        objectsInArrays: true,
        singleValue: true,
      },
    ],
    'vue/arrow-spacing': [
      'error',
      {
        after: true,
        before: true,
      },
    ],
    'vue/block-spacing': ['error', 'always'],
    'vue/brace-style': [
      'error',
      '1tbs',
      {
        allowSingleLine: false,
      },
    ],
    'vue/camelcase': [
      'error',
      {
        ignoreDestructuring: false,
        ignoreGlobals: false,
        ignoreImports: false,
        properties: 'always',
      },
    ],
    'vue/comma-dangle': ['error', 'always-multiline'],
    'vue/comma-spacing': [
      'error',
      {
        after: true,
        before: false,
      },
    ],
    'vue/comma-style': ['error', 'last'],
    'vue/dot-location': ['error', 'property'],
    'vue/dot-notation': [
      'error',
      {
        allowKeywords: false,
      },
    ],
    'vue/eqeqeq': ['error', 'always'],
    'vue/func-call-spacing': ['error', 'never'],
    'vue/key-spacing': [
      'error',
      {
        afterColon: true,
        align: 'value',
        beforeColon: false,
        mode: 'strict',
      },
    ],
    'vue/keyword-spacing': [
      'error',
      {
        after: true,
        before: true,
      },
    ],
    'vue/max-len': [
      'error',
      {
        code: 128,
        comments: 128,
        ignoreComments: false,
        ignoreHTMLAttributeValues: false,
        ignoreHTMLTextContents: false,
        ignorePattern: '',
        ignoreRegExpLiterals: false,
        ignoreStrings: false,
        ignoreTemplateLiterals: false,
        ignoreTrailingComments: false,
        ignoreUrls: false,
        tabWidth: 2,
        template: 128,
      },
    ],
    'vue/multiline-ternary': ['error', 'always-multiline'],
    'vue/no-constant-condition': 'error',
    'vue/no-empty-pattern': 'error',
    'vue/no-extra-parens': 'error',
    'vue/no-irregular-whitespace': [
      'error',
      {
        skipComments: false,
        skipHTMLAttributeValues: false,
        skipHTMLTextContents: false,
        skipRegExps: false,
        skipStrings: true,
        skipTemplates: false,
      },
    ],
    'vue/no-loss-of-precision': 'error',
    'vue/no-restricted-syntax': 'off',
    'vue/no-sparse-arrays': 'error',
    'vue/no-useless-concat': 'error',
    'vue/object-curly-newline': [
      'error',
      {
        consistent: true,
        multiline: true,
      },
    ],
    'vue/object-curly-spacing': [
      'error',
      'always',
      {
        arraysInObjects: false,
        objectsInObjects: false,
      },
    ],
    'vue/object-property-newline': 'off',
    'vue/object-shorthand': [
      'error',
      'always',
      {
        avoidExplicitReturnArrows: true,
        avoidQuotes: true,
        ignoreConstructors: true,
      },
    ],
    'vue/operator-linebreak': ['error', 'before'],
    'vue/prefer-template': 'error',
    'vue/quote-props': [
      'error',
      'as-needed',
      {
        keywords: true,
        numbers: true,
        unnecessary: true,
      },
    ],
    'vue/space-in-parens': ['error', 'never'],
    'vue/space-infix-ops': [
      'error',
      {
        int32Hint: false,
      },
    ],
    'vue/space-unary-ops': [
      'error',
      {
        nonwords: true,
        words: true,
      },
    ],
    'vue/template-curly-spacing': ['error', 'always'],
  },
}


src/vue2.js
/*
 * @Date: 2022-12-12 23:09:35
 * @Author: liting luz.liting@gmail.com
 * @LastEditors: liting luz.liting@gmail.com
 * @LastEditTime: 2022-12-13 22:59:04
 * @FilePath: /eslint-config-liting/src/vue2.js
 */
module.exports = {
  'vue/block-lang': [
    'error',
    {
      script: {
        lang: 'js',
      },
    },
  ],
  'vue/component-api-style': ['error', ['options']],
  'vue/no-deprecated-destroyed-lifecycle': 'off',
  'vue/no-deprecated-dollar-listeners-api': 'off',
  'vue/no-deprecated-dollar-scopedslots-api': 'off',
  'vue/no-deprecated-events-api': 'off',
  'vue/no-deprecated-filter': 'off',
  'vue/no-deprecated-functional-template': 'off',
  'vue/no-deprecated-inline-template': 'off',
  'vue/no-deprecated-props-default-this': 'off',
  'vue/no-deprecated-router-link-tag-prop': [
    'warn',
    {
      components: ['RouterLink'],
    },
  ],
  'vue/no-deprecated-v-bind-sync': 'off',
  'vue/no-deprecated-v-on-native-modifier': 'off',
  'vue/no-deprecated-v-on-number-modifiers': 'off',
  'vue/no-deprecated-vue-config-keycodes': 'off',
  'vue/no-expose-after-await': 'off',
  'vue/no-lifecycle-after-await': 'off',
  'vue/no-unsupported-features': [
    'error',
    {
      version: '^2.6.0',
    },
  ],
  'vue/no-v-for-template-key-on-child': 'off',
  'vue/no-watch-after-await': 'off',
  'vue/require-emit-validator': 'off',
  'vue/require-slots-as-functions': 'off',
  'vue/v-for-delimiter-style': ['error', 'in'],

  // essential for Vue.js 2.x
  // eslint-disable-next-line
  'vue/no-custom-modifiers-on-v-model': 'error',
  'vue/no-multiple-template-root': 'error',
  'vue/no-v-for-template-key': 'error',
  'vue/no-v-model-argument': 'error',
  'vue/valid-model-definition': 'error',
  'vue/valid-v-bind-sync': 'error',
}


src/ts.js
/*
 * @Date: 2022-12-14 22:45:22
 * @Author: liting luz.liting@gmail.com
 * @LastEditors: liting luz.liting@gmail.com
 * @LastEditTime: 2022-12-17 14:55:33
 * @FilePath: /eslint-config-liting/src/ts.js
 */
module.exports = {
  overrides: [
    {
      extends: ['plugin:@typescript-eslint/eslint-recommended'],
      files: ['*.ts'],
      parser: '@typescript-eslint/parser',
      plugins: ['@typescript-eslint'],
      /*
       * rules: {
       *   // typeScript rules
       *   '@typescript-eslint/adjacent-overload-signatures': 'error',
       *   '@typescript-eslint/array-type': [
       *     'error',
       *     {
       *       default: 'array',
       *       readonly: 'array',
       *     },
       *   ],
       *   '@typescript-eslint/await-thenable': 'error',
       *   '@typescript-eslint/ban-ts-comment': [
       *     'error',
       *     {
       *       'ts-check': false,
       *       'ts-expect-error': false,
       *       'ts-ignore': true,
       *       'ts-nocheck': 'allow-with-description',
       *     },
       *   ],
       *   '@typescript-eslint/ban-tslint-comment': 'warn',
       *   '@typescript-eslint/ban-types': 'error',
       *   '@typescript-eslint/class-literal-property-style': ['error', 'fields'],
       *   '@typescript-eslint/consistent-generic-constructors': ['error', 'type-annotation'],
       *   '@typescript-eslint/consistent-indexed-object-style': ['error', 'record'],
       *   '@typescript-eslint/consistent-type-assertions': [
       *     'error',
       *     {
       *       assertionStyle: 'as',
       *       objectLiteralTypeAssertions: 'never',
       *     },
       *   ],
       *   '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
       *   '@typescript-eslint/consistent-type-exports': [
       *     'error',
       *     {
       *       fixMixedExportsWithInlineTypeSpecifier: false,
       *     },
       *   ],
       *   '@typescript-eslint/consistent-type-imports': [
       *     'error',
       *     {
       *       disallowTypeAnnotations: true,
       *       fixStyle: 'separate-type-imports',
       *       prefer: 'type-imports',
       *     },
       *   ],
       *   '@typescript-eslint/explicit-function-return-type': [
       *     'error',
       *     {
       *       allowConciseArrowFunctionExpressionsStartingWithVoid: true,
       *       allowDirectConstAssertionInArrowFunctions: true,
       *       allowExpressions: true,
       *       allowHigherOrderFunctions: true,
       *       allowTypedFunctionExpressions: true,
       *     },
       *   ],
       *   '@typescript-eslint/explicit-member-accessibility': [
       *     'error',
       *     {
       *       accessibility: 'no-public',
       *     },
       *   ],
       *   '@typescript-eslint/explicit-module-boundary-types': [
       *     'error',
       *     {
       *       allowArgumentsExplicitlyTypedAsAny: false,
       *       allowDirectConstAssertionInArrowFunctions: false,
       *       allowHigherOrderFunctions: false,
       *       allowTypedFunctionExpressions: false,
       *     },
       *   ],
       *   '@typescript-eslint/member-ordering': [
       *     'error',
       *     {
       *       default: [
       *         // index signature
       *         'signature',
       *         'call-signature',
       */

      /*
       *         // fields
       *         'public-static-field',
       *         'protected-static-field',
       *         'private-static-field',
       *         '#private-static-field',
       */

      /*
       *         'public-decorated-field',
       *         'protected-decorated-field',
       *         'private-decorated-field',
       */

      /*
       *         'public-instance-field',
       *         'protected-instance-field',
       *         'private-instance-field',
       *         '#private-instance-field',
       */

      /*
       *         'public-abstract-field',
       *         'protected-abstract-field',
       */

      /*
       *         'public-field',
       *         'protected-field',
       *         'private-field',
       *         '#private-field',
       */

      /*
       *         'static-field',
       *         'instance-field',
       *         'abstract-field',
       */

      //         'decorated-field',

      //         'field',

      /*
       *         // static initialization
       *         'static-initialization',
       */

      /*
       *         // constructors
       *         'public-constructor',
       *         'protected-constructor',
       *         'private-constructor',
       */

      //         'constructor',

      /*
       *         // getters
       *         'public-static-get',
       *         'protected-static-get',
       *         'private-static-get',
       *         '#private-static-get',
       */

      /*
       *         'public-decorated-get',
       *         'protected-decorated-get',
       *         'private-decorated-get',
       */

      /*
       *         'public-instance-get',
       *         'protected-instance-get',
       *         'private-instance-get',
       *         '#private-instance-get',
       */

      /*
       *         'public-abstract-get',
       *         'protected-abstract-get',
       */

      /*
       *         'public-get',
       *         'protected-get',
       *         'private-get',
       *         '#private-get',
       */

      /*
       *         'static-get',
       *         'instance-get',
       *         'abstract-get',
       */

      //         'decorated-get',

      //         'get',

      /*
       *         // setters
       *         'public-static-set',
       *         'protected-static-set',
       *         'private-static-set',
       *         '#private-static-set',
       */

      /*
       *         'public-decorated-set',
       *         'protected-decorated-set',
       *         'private-decorated-set',
       */

      /*
       *         'public-instance-set',
       *         'protected-instance-set',
       *         'private-instance-set',
       *         '#private-instance-set',
       */

      /*
       *         'public-abstract-set',
       *         'protected-abstract-set',
       */

      /*
       *         'public-set',
       *         'protected-set',
       *         'private-set',
       *         '#private-set',
       */

      /*
       *         'static-set',
       *         'instance-set',
       *         'abstract-set',
       */

      //         'decorated-set',

      //         'set',

      /*
       *         // methods
       *         'public-static-method',
       *         'protected-static-method',
       *         'private-static-method',
       *         '#private-static-method',
       */

      /*
       *         'public-decorated-method',
       *         'protected-decorated-method',
       *         'private-decorated-method',
       */

      /*
       *         'public-instance-method',
       *         'protected-instance-method',
       *         'private-instance-method',
       *         '#private-instance-method',
       */

      /*
       *         'public-abstract-method',
       *         'protected-abstract-method',
       */

      /*
       *         'public-method',
       *         'protected-method',
       *         'private-method',
       *         '#private-method',
       */

      /*
       *         'static-method',
       *         'instance-method',
       *         'abstract-method',
       */

      //         'decorated-method',

      /*
       *         'method',
       *       ],
       *     },
       *   ],
       *   '@typescript-eslint/method-signature-style': ['error', 'property'],
       *   '@typescript-eslint/naming-convention': [
       *     'warn',
       *     {
       *       format: ['camelCase'],
       *       leadingUnderscore: 'allow',
       *       selector: 'default',
       *       trailingUnderscore: 'allow',
       *     },
       */

      /*
       *     {
       *       format: ['camelCase', 'UPPER_CASE'],
       *       leadingUnderscore: 'allow',
       *       selector: 'variable',
       *       trailingUnderscore: 'allow',
       *     },
       */

      /*
       *     {
       *       format: ['PascalCase'],
       *       selector: 'typeLike',
       *     },
       *   ],
       *   '@typescript-eslint/no-base-to-string': 'error',
       *   '@typescript-eslint/no-confusing-non-null-assertion': 'error',
       *   '@typescript-eslint/no-confusing-void-expression': [
       *     'error',
       *     {
       *       ignoreArrowShorthand: true,
       *       ignoreVoidOperator: true,
       *     },
       *   ],
       *   '@typescript-eslint/no-duplicate-enum-values': 'error',
       *   '@typescript-eslint/no-dynamic-delete': 'error',
       *   '@typescript-eslint/no-empty-interface': 'error',
       *   '@typescript-eslint/no-explicit-any': 'error',
       *   '@typescript-eslint/no-extra-non-null-assertion': 'error',
       *   '@typescript-eslint/no-extraneous-class': [
       *     'error',
       *     {
       *       allowConstructorOnly: true,
       *       allowEmpty: false,
       *       allowStaticOnly: false,
       *       allowWithDecorator: true,
       *     },
       *   ],
       *   '@typescript-eslint/no-floating-promises': [
       *     'error',
       *     {
       *       ignoreIIFE: true,
       *       ignoreVoid: true,
       *     },
       *   ],
       *   '@typescript-eslint/no-for-in-array': 'error',
       *   '@typescript-eslint/no-inferrable-types': [
       *     'error',
       *     {
       *       ignoreParameters: true,
       *       ignoreProperties: true,
       *     },
       *   ],
       *   '@typescript-eslint/no-invalid-void-type': [
       *     'error',
       *     {
       *       allowAsThisParameter: false,
       *       allowInGenericTypeArguments: false,
       *     },
       *   ],
       *   '@typescript-eslint/no-meaningless-void-operator': [
       *     'error',
       *     {
       *       checkNever: false,
       *     },
       *   ],
       *   '@typescript-eslint/no-misused-new': 'error',
       *   '@typescript-eslint/no-misused-promises': [
       *     'error',
       *     {
       *       checksConditionals: true,
       *       checksSpreads: true,
       *       checksVoidReturn: true,
       *     },
       *   ],
       *   '@typescript-eslint/no-namespace': [
       *     'error',
       *     {
       *       allowDeclarations: true,
       *       allowDefinitionFiles: true,
       *     },
       *   ],
       *   '@typescript-eslint/no-non-null-asserted-nullish-coalescing': 'error',
       *   '@typescript-eslint/no-non-null-asserted-optional-chain': 'error',
       *   '@typescript-eslint/no-non-null-assertion': 'error',
       *   '@typescript-eslint/no-redundant-type-constituents': 'error',
       *   '@typescript-eslint/no-require-imports': 'warn',
       *   '@typescript-eslint/no-this-alias': [
       *     'warn',
       *     {
       *       allowDestructuring: true,
       *     },
       *   ],
       *   '@typescript-eslint/no-type-alias': [
       *     'error',
       *     {
       *       allowAliases: 'in-unions-and-intersections',
       *       allowCallbacks: 'always',
       *       allowConditionalTypes: 'always',
       *       allowConstructors: 'always',
       *       allowLiterals: 'in-unions-and-intersections',
       *       allowMappedTypes: 'in-unions-and-intersections',
       *       allowTupleTypes: 'in-unions-and-intersections',
       *     },
       *   ],
       * },
       */
    },
  ],
}
